packages/compass-editor/src/json-editor.tsx

```diff
@@ -866,7 +869,7 @@ const BaseEditor = React.forwardRef<EditorRef, EditorProps>(function BaseEditor(
       const isSyntaxTreeAvailable =
         syntaxTreeAvailable(editor.state, docLength) ||
         forceParsing(editor, docLength, 150);
-      if (isSyntaxTreeAvailable) {
+      if (initialJSONFoldAll.current && isSyntaxTreeAvailable) {
```

Comment:
Not a blocker, but a question – I have to admit I don’t have the easiest time following this :)

We’re using `useRef()` here to keep track of the value of `initialJSONFoldAll` that was seen when the `BaseEditor` component was first rendered, right? I can understand how that’s different behavior from just using `initialJSONFoldAll` directly _if_ there is some parent component which passes a `initialJSONFoldAll` value that changes over time. But given that the default value is `true`, and we’re only passing it from one additional component that’s passing constant `false` – how _does_ this differ from not using `useRef` in practice?

Reply:
A small difference is that eslint rules for hooks can identify ref values and so you don't get warnings for missing effect dependencies when using them. We can use `initialJSONFoldAll` here directly, but then we'd need to disable the rule for this effect. If you'd have a preference for that, I can do this change in a follow up patch

Reply:
No, not an issue – just something that was a bit unclear to me. I’d maybe consider leaving a short comment, but feel free to merge this as-is.